build/BranchInfo.props

```diff
@@ -1,5 +1,28 @@
 <Project>
   <PropertyGroup>
+    <StableProjects>
+      Microsoft.ML;
+      Microsoft.ML.Core;
+      Microsoft.ML.CpuMath;
+      Microsoft.ML.Data;
+      Microsoft.ML.FastTree;
+      Microsoft.ML.ImageAnalytics;
+      Microsoft.ML.KMeansClustering;
+      Microsoft.ML.LightGbm;
+      Microsoft.ML.PCA;
+      Microsoft.ML.StandardTrainers;
+      Microsoft.ML.Transforms;
+      Microsoft.ML.Mkl.Components;
+      Microsoft.ML.Mkl.Redist;
+    </StableProjects>
+    <IsStableProject Condition="'$(MSBuildProjectName.Contains(.symbols))' == 'false'">$(StableProjects.Contains($(MSBuildProjectName)))</IsStableProject>
+    <IsStableProject Condition="'$(MSBuildProjectName.Contains(.symbols))' == 'true'">$(StableProjects.Contains($(MSBuildProjectName.Substring(0, $(MSBuildProjectName.IndexOf(.symbols))))))</IsStableProject>
+    <MajorVersion>0</MajorVersion>
```

Comment:
It may make more sense to split these properties into their own PropertyGroup. Then it will be easier to read which are "stable" and which are "preview". Along with that, I think `UseStableVersionForNativeAssets` should just be rolled into the `IsStableProject` condition. So something like this:

```xml
<Project>
  <PropertyGroup>
    <StableProjects>
      Microsoft.ML;
      Microsoft.ML.Core;
      Microsoft.ML.CpuMath;
      Microsoft.ML.Data;
      Microsoft.ML.FastTree;
      Microsoft.ML.ImageAnalytics;
      Microsoft.ML.KMeansClustering;
      Microsoft.ML.LightGbm;
      Microsoft.ML.PCA;
      Microsoft.ML.StandardTrainers;
      Microsoft.ML.Transforms;
      Microsoft.ML.Mkl.Components;
      Microsoft.ML.Mkl.Redist;
    </StableProjects>
    <IsStableProject Condition="'$(UseStableVersionForNativeAssets)' == 'true'">true</IsStableProject>
    <IsStableProject Condition="'$(IsStableProject)' == '' AND '$(MSBuildProjectName.Contains(.symbols))' == 'false'">$(StableProjects.Contains($(MSBuildProjectName)))</IsStableProject>
    <IsStableProject Condition="'$(IsStableProject)' == '' AND '$(MSBuildProjectName.Contains(.symbols))' == 'true'">$(StableProjects.Contains($(MSBuildProjectName.Substring(0, $(MSBuildProjectName.IndexOf(.symbols))))))</IsStableProject>
  </PropertyGroup>

  <PropertyGroup Condition="'$(IsStableProject)' == 'true'
    <MajorVersion>0</MajorVersion>
    <MinorVersion>12</MinorVersion>
    <PatchVersion>0</PatchVersion>
    <PreReleaseLabel>preview</PreReleaseLabel>
  </PropertyGroup>
  <PropertyGroup Condition="'$(IsStableProject)' != 'true'">
    <MajorVersion>0</MajorVersion>
    <MinorVersion>12</MinorVersion>
    <PatchVersion>0</PatchVersion>
    <PreReleaseLabel>preview</PreReleaseLabel>
  </PropertyGroup>
</Project>
```